docs/diagrams/OptimiseSequenceDiagram.puml

```diff
@@ -0,0 +1,82 @@
+@startuml
+!include style.puml
+
+box Logic LOGIC_COLOR_T1
+participant ":LogicManager" as LogicManager LOGIC_COLOR
+participant ":PlannerParser" as PlannerParser LOGIC_COLOR
+participant ":OptimiseCommandParser" as OptimiseCommandParser LOGIC_COLOR
+participant ":OptimiseBudgetCommand" as OptimiseBudgetCommand LOGIC_COLOR
+participant ":CommandResult" as CommandResult LOGIC_COLOR
```

Comment:
CommandResult might not be necessary for this sequence diagram; it's extra detail that does not really affect the understanding of how optimisation works